### ISSUE TITLE
new rule: enforce `<script setup>`  over regular `<script>`

### ISSUE THREAD

**Issue by mesqueeb** [vuejs/eslint-plugin-vue#1971]
**Please describe what the rule should do:**

I would love to have an eslint rule that will make sure all my components use
- `<script lang="ts" setup>`
- and NOT
- `<script setup>`
- and also NOT
- `<script>`

**What category should the rule belong to?**

[x] Enforces code style (layout)
[ ] Warns about a potential error (problem)
[ ] Suggests an alternate way of doing something (suggestion)
[ ] Other (please specify:)

**example**

GOOD
```vue
<script lang="ts" setup>
  // ...
```

GOOD
```vue
<script lang="ts">
  // ... a component with an _extra_ script tag is also OK
</script>

<script lang="ts" setup>
  // ...
```

BAD
```vue
<script setup>
  // ...
```

BAD
```vue
<script>
  // ...
```

**Comment by mesqueeb** [comment]
It's a bit hard to find, but this can be achieved with:
```
"vue/component-api-style": ["error", ["script-setup"]],
"vue/block-lang": ["error", { "script": { "lang": "ts" } }],
```
